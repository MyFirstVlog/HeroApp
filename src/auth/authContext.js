import { createContext } from "react/cjs/react.development";


export const AuthContext = createContext(); //? Es un componente, debe estar en lo mas alto de la aplicaci√≥n