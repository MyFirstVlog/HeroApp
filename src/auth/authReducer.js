//? reducedr es una funcion

import { Types } from "../types/Types";

// const state = {
//     name: 'Alejandro',
//     logged: true
// }

/*
Asi luce laaccion que activara al reducer
const loginAction = {
    type: Types.login,
    payload: {
        name: 'Alejandro',
        email: 'alejoestradam@yahoo.com'
    }
}

*/
export const authReducer = (state = {}, action) => { //? Accion que modifica el state, fns puras que no tioenen que salirse de su funcion para cumplir el trabajo

    switch (action.type) {
        case Types.login:
            return{
                ...action.payload,
                logged: true
            }
        case Types.logout:
            return{
                logged: false
            }
            
        default:
            return state;
    }

}