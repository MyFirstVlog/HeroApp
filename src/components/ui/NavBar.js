import React, { useContext } from 'react'
import { useNavigate, useParams } from 'react-router'
import { Link, NavLink, Navigate } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext'
import { types } from '../../types/types'



export const NavBar = () => {
    const {user:{name}, dispatch} = useContext(AuthContext)

    let history = useNavigate()

    const handleLogout = () => {

        dispatch({
            type:types.logout,
            payload: {
                logged : false
            }
        })

        history('/login',{replace:true})

        
    }

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            
            <Link 
                className="navbar-brand" 
                to="/"
            >
                Asociaciones
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/dc"
                    >
                        DCs
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/search"
                    >
                        Searh Hero
                    </NavLink>
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
                <ul className="navbar-nav ml-auto">
                    <span
                        className="nav-item nav-link text-info"
                    >
                        {name}
                    </span>
                    <button
                        className="nav-item nav-link btn" 
                        onClick={handleLogout}
                    >
                        Logout
                    </button> 
                </ul>
            </div>
        </nav>
    )
}

 