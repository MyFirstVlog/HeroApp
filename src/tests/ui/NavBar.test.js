import { mount } from "enzyme"
import React from 'react';
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { NavBar } from "../../components/ui/NavBar";

const handleLogout = jest.fn()

const contextValue = {
    dispatch : jest.fn(),
    user:{
        logged:true,
        name : 'Alejandro'
    }

}
describe('Pruebas en <NavBar/>', () => {

    const historyMock = {}
    
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        </AuthContext.Provider>
    )

    test('debe de mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('span').text().trim()).toBe(contextValue.user.name)
    })

    test('debe de llamar el logout y el usar history ', () => {
        wrapper.find('button').simulate('click')
        expect(contextValue.dispatch).toHaveBeenCalledWith({payload:{
                                                                logged: false,
                                                            },
                                                            type: "[auth] logout",})
    })


    
    
})
