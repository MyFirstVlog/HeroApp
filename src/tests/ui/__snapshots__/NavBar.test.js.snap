// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NavBar/> debe de mostrarse correctamente  1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        activeclassname="active"
        className="nav-item nav-link"
        exact={true}
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        activeclassname="active"
        className="nav-item nav-link"
        exact={true}
        href="/dc"
        onClick={[Function]}
      >
        DCs
      </a>
      <a
        activeclassname="active"
        className="nav-item nav-link"
        exact={true}
        href="/search"
        onClick={[Function]}
      >
        Searh Hero
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Alejandro
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
